File: Backend/controllers/evaluation.js
const Evaluation = require('../models/evaluation');

const onCreateEvaluation = async (req, res) => {
  try {

    
    const { term ,sectionId } = req.params;
    const { courseMessage, instructorMessage } = req.body;
    const evaluation = await Evaluation.createEvaluation(term, sectionId, courseMessage, instructorMessage);
    return res.status(200).json({ success: true, evaluation });
  } catch (error) {
    console.error(error);
    return res.status(500).json({ success: false, error: error.message });
  }
};

const onGetAllEvaluationsBySectionId = async (req, res) => {
  try {
    const { term, sectionId } = req.params;
    const evaluations = await Evaluation.getAllEvaluationsBySectionId(term, sectionId);
    return res.status(200).json({ success: true, evaluations });
  } catch (error) {
    console.error(error);
    return res.status(500).json({ success: false, error: error.message });
  }
};

module.exports = {
  onCreateEvaluation,
  onGetAllEvaluationsBySectionId,
};

File: Frontend/lib/Objects/evaluation.dart
class Evaluation {
  String sectionId;
  String courseMessage;
  String instructorMessage;

  Evaluation({
    required this.sectionId,
    this.courseMessage = '',
    this.instructorMessage = '',
  });

  Map<String, dynamic> toJson() {
    return {
      'sectionId': sectionId,
      'courseMessage': courseMessage,
      'instructorMessage': instructorMessage,
    };
  }

  factory Evaluation.fromJson(Map<String, dynamic> json) {
    return Evaluation(
      sectionId: json['sectionId'],
      courseMessage: json['courseMessage'] ?? '',
      instructorMessage: json['instructorMessage'] ?? '',
    );
  }
}
